Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Lift.java

```diff
@@ -1,77 +1,89 @@
 package org.usfirst.frc.team199.Robot2018.subsystems;
 
+import org.usfirst.frc.team199.Robot2018.OI;
 import org.usfirst.frc.team199.Robot2018.RobotMap;
+import org.usfirst.frc.team199.Robot2018.commands.AutoLift;
 import org.usfirst.frc.team199.Robot2018.subsystems.LiftInterface.Position;
 
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
 
 import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Command;
 import edu.wpi.first.wpilibj.command.Subsystem;
 
 /**
  *
  */
 public class Lift extends Subsystem implements LiftInterface {
 
-	private final WPI_TalonSRX liftMotor = RobotMap.liftMotor;
+	private final SpeedControllerGroup liftMotors = RobotMap.liftMotors;
 	private final Encoder liftEnc = RobotMap.liftEnc;
-	private Position targetPosition = Position.GROUND;
+	private Position currPosition = Position.GROUND;
 	
 	/**
 	 * Set the default command for a subsystem here.
 	 * */
     public void initDefaultCommand() {
         // Set the default command for a subsystem here.
         //setDefaultCommand(new MySpecialCommand());
+    	int ang = OI.manipulator.getPOV();
+    	Command com = null;
+    	if(ang == 0)
+    		com = new AutoLift(Position.GROUND, this);
+    	else if(ang == 90)
+    		com = new AutoLift(Position.SWITCH, this);
+    	else if(ang == 180)
+    		com = new AutoLift(Position.SWITCH, this);
+    	else if(ang == 270)
+    		com = new AutoLift(Position.SWITCH, this);
+    	if(com != null)
+    		setDefaultCommand(com);
```

Comment:
That makes more sense, what do you recommend I use to achieve that? Is there a way to call AutoLift from another command so that I don't repeat functionality?

Reply:
Here's what I'd do:
1. Move the PID control of the lift out of the commands. It should always be running to move the lift to the desired height.
2. Make the AutoLift() command just set the target height and wait until it is reached.
3. Make the DefaultLift() command set the target height based on POV angle (and perhaps other input).